hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -905,25 +909,32 @@ public void onUpdatePartitionColumnStat(UpdatePartitionColumnStatEvent updatePar
   }
 
   @Override
-  public void onUpdatePartitionColumnStatDirectSql(UpdatePartitionColumnStatEvent updatePartColStatEvent,
-                                                   Connection dbConn, SQLGenerator sqlGenerator)
+  public void onUpdatePartitionColumnStatInBatch(UpdatePartitionColumnStatEventBatch updatePartColStatEventBatch,
```

Comment:
Can this be used for updatePartitionColStatsForOneBatch as well in HMSHandler

Reply:
updatePartitionColStatsInBatch internally uses updatePartitionColStatsForOneBatch method.updatePartitionColStatsForOneBatch calls direct sql method for one batch and that internally writes the notification log.

Reply:
updatePartitionColStatsForOneBatch calls the direct sql method but also does the following MetaStoreListenerNotifier.notifyEvent(listeners,
                EventMessage.EventType.UPDATE_PARTITION_COLUMN_STAT,
                new UpdatePartitionColumnStatEvent(colStats, partVals, parameters,
                        tbl, writeId, this));
for each event.

Reply:
Yes ..for normal listeners we can not change as they may be expecting the data in that way. The change is done only for transactional listeners (DBNotification listener is a transactional listener). The notification for transactional listeners are done inside the direct sql method  as it has to be within same transaction. For normal listeners we need not have them in the same transaction.